### PR TITLE
[PP-2286] adding product_viewed logic to product_clicked to make sure a clicked product is always viewed as well

### PR DESCRIPTION
**Pull request by resutovsheryan**

### CODE REVIEW

**Review by raymondweidner — APPROVED** [review]
Looks like it will work, but I have a small nit. Can we break some of this code into functions to avoid repetition?

File: braze-product-script.js [hunk 1]
```diff
@@ -41,14 +41,22 @@
...
-			}      
+			}
+
+			const alreadyViewedProducts = JSON.parse(localStorage.getItem('viewed_products') || "{}");
+			productLastViewedAt = alreadyViewedProducts[productName] || 0
+			if(now-productLastViewedAt > one_day_millis){
+				alreadyViewedProducts[productName] = now;
+				fetchAndLogProduct("/products/"+productName, 'shopify_product_viewed');
+			}
+			localStorage.setItem('viewed_products', JSON.stringify(alreadyViewedProducts));
```
Comment (lexholden) [inline braze-product-script.js]: Since 'viewed_products' is used multiple times, why not define it as a constant?

Comment (lexholden) [reply braze-product-script.js]: Also, we don't need to call setItem unless we made updates, surely? Can this line not be inside the if?

**Review by lexholden — APPROVED** [review]
Some nits, but LGTM

Though +1 on Ray's suggestion to refactor away the copy paste here, seems there could be some sort of `fetchAndLogProductIf` that takes the product name and a condition and handles the localStorage update

File: braze-product-script.js [hunk 2]
```diff
@@ -33,22 +33,26 @@
 		return Object.fromEntries(urlSearch.entries())
 	}
 
+	function attempt_to_log_event(event, productPath, cachedProductsStr, now, oneDayMillis) {
```
Comment (lexholden) [inline braze-product-script.js:36]: Inconsistent camelCase and snake_case here....?

File: braze-product-script.js (hunk 1 above)
Comment (resutovsheryan) [reply braze-product-script.js]: Yea, that was originally, there but there was some weird issue i believe with the observer that was resulting in inconsistent behavior. Trying to remember what it was...

**Pull request merged**